src/librustc_typeck/check/mod.rs

```diff
@@ -5086,64 +5097,56 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
     /// Report errors if the provided parameters are too few or too many.
     fn check_generic_arg_count(&self,
```

Comment:
Oh, I guess you still use separate checking from iterating parameters and arguments in lockstep?

Reply:
For now. As always, I wasn't sure whether this would be better addressed now or later. I'll look into it.